### PR TITLE
Implementing shopping cart

### PR DESCRIPTION
**Pull request by BohdanTrue**
create `ShoppingCartController ` and some points in it:
`GET: /api/books` - get info about users shopping cart
`POST: /api/books` - add an item to shopping cart
`PUT: /api/books/cart-items/{itemId}` - update quantity of item in shopping cart
`DELETE: /api/books/cart-items/{itemId}` - delete an item from shopping cart

### CODE REVIEW

**Review by DevIgork — APPROVED** [review]
LGTM

File: src/main/java/mate/academy/bookstore/model/CartItem.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+@SQLDelete(sql = "UPDATE cart_items c SET c.is_deleted = true WHERE c.id=?")
+@Where(clause = "is_deleted = false")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/CartItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY)
```

File: src/main/java/mate/academy/bookstore/model/CartItem.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    @JoinColumn(name = "shopping_cart_id")
+    private ShoppingCart shoppingCart;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/CartItem.java]:
```suggestion
    @ManyToOne(fetch = FetchType.LAZY)
```

File: src/main/java/mate/academy/bookstore/model/ShoppingCart.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@SQLDelete(sql = "UPDATE shopping_carts s SET s.is_deleted = true WHERE s.id=?")
+@Where(clause = "is_deleted = false")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @OneToOne
```
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/ShoppingCart.java]:
```suggestion
    @OneToOne(fetch = FetchType.LAZY)
```

File: src/main/java/mate/academy/bookstore/repository/book/BookSpecificationBuilder.java
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/repository/book/BookSpecificationBuilder.java:1]: this change not related to this hw.

File: src/main/java/mate/academy/bookstore/model/Book.java
Comment (d1sam) [inline src/main/java/mate/academy/bookstore/model/Book.java:1]: this change not related to this hw.

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but there are some changes should be done!

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**